google/cloud/internal/default_completion_queue_impl.cc

```diff
@@ -96,7 +97,7 @@ class AsyncTimerFuture : public AsyncGrpcOperation {
   }
 
   static ValueType Cancelled() {
-    return Status{StatusCode::kCancelled, "timer canceled"};
+    return internal::CancelledError("timer canceled", GCP_ERROR_INFO());
```

Comment:
nit: maybe it should receive the error info as a parameter.  OTOH, there is a single call site, maybe time to remove the function altogether.